tools/mason/MasonNew.chpl

```diff
@@ -55,12 +51,48 @@ proc masonNew(args) {
         name = arg;
       }
     }
-    InitProject(name, vcs, show);
+
+    if name == '' {
+      writeln("No package name specified");
+    } else if !isIdentifier(name) {
+      writeln("Bad package name '", name,
+               "' - only Chapel identifiers are legal package names");
+    } else if name.count("$") > 0 {
+      writeln("Bad package name '", name,
+              "' - $ is not allowed in package names");
+    } else if isDir(name) {
+      writeln("A directory named '", name, "' already exists");
+    } else {
+      InitProject(name, vcs, show);
+    }
   }
 }
 
-
-
+proc isIdentifier(name:string) {
```

Comment:
I'd be in favor of a simple doc-string clarifying this function, e.g.

```
/* Return 'true' for valid identifiers according to Chapel spec, otherwise 'false' */
```